xls/modules/rle/rle.x

```diff
@@ -0,0 +1,253 @@
+// Copyright 2023 The XLS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file implements a parametric RLE encoder
+//
+// The encoder uses Run Length Encoding (RLE) to compress the input stream
+// with symbols (data to compress). The output contains packets of data with
+// the symbols and the number of their occurrences in the input stream.
+
+import std
+
+// structure containing output data from an RLE encoder
+struct EncOutData<SYMB_WIDTH: u32, CNT_WIDTH: u32> {
+    symb: bits[SYMB_WIDTH], // symbol
+    cnt: bits[CNT_WIDTH],   // symbol counter
+}
+
+// structure to preserve the state of an RLE encoder for the next iteration
+pub struct RLEEncState<SYMB_WIDTH: u32, CNT_WIDTH: u32> {
+    prev: EncOutData<SYMB_WIDTH, CNT_WIDTH>, // previous symbol and symbol counter
```

Comment:
I think we could simply name this `count`?

Reply:
Renamed to `COUNT_WIDTH`

Reply:
Sorry about the confusion, I was referring the `prev` field:  describe it as something used to track the count for the current symbol (`counter`) rather that `prev` might be more informative.